models/Document/PageSnippet.php

```diff
@@ -604,6 +606,7 @@ public function getElements()
     public function getEditables(): array
     {
         if ($this->editables === null) {
+            $this->editables = [];
```

Comment:
This shouldn't be necessary since `$this->getDao()->getEditables()` always returns an array, right?

Reply:
Yes, but since $this->editables is not initialized (i.e. null) and getEditables calls setEditable this will result in an endless recursion otherwise. See https://github.com/pimcore/pimcore/pull/6991#issue-473165647

Reply:
No, it doesn't call `setEditable` but `setEditables` which just sets the field. I don't see why this should result in an endless loop.

Reply:
@weisswurstkanone is right, with this PR `setEditable()` is again calling `getEditables()` which would result in an endless loop.

Reply:
But I'm wondering why this is happening, ... it should be sufficient if `\Pimcore\Model\Document\PageSnippet\Dao::getEditables()` just returns the editables ... calling `setEditable()` first and then calling `PageSnippet::setEditables()` with the return value of `Dao::getEditables()` sounds a bit weird to me.

Reply:
why not initialise editables as an empty array? otherwise the strict type change will fail here anyway